dart/deepgram_transcribe_video/lib/main.dart

```diff
@@ -0,0 +1,64 @@
+import 'dart:convert';
+import 'package:http/http.dart' as http;
+
+//custom success message
+void returnSuccess(final res, final data) {
+  res.json({
+    'success': true,
+    'deepgramData': data,
+  });
+}
+
+//custom failure message
+void returnFailure(final res, final message) {
+  res.json({
+    'success': false,
+    'message': message,
+  });
+}
+
+Future<void> start(final req, final res) async {
+  var fileUrl;
+  var APIkey;
+
+  //getting data from payload
+  try {
+    final data = jsonDecode(req.payload);
+    fileUrl = data["fileUrl"];
+    APIkey = req.env["DEEPGRAM_API_KEY"];
+  } catch (err) {
+    returnFailure(res, err.toString());
+    return;
+  }
+
+  //checks
+  if (fileUrl == null || APIkey == null) {
+    if (fileUrl == null)
+      returnFailure(res, "Provide a valid file URL in payload");
+    else
+      returnFailure(res, "Provide the API key as an environment variable");
+    return;
+  }
```

Comment:
There's a style of programming called "return early" to improve readability by cutting down on nested code. Using that pattern, the code would look like:

```suggestion

  if (fileUrl == null) {
    returnFailure(res, "Provide a valid file URL in payload");
    return;
  }
  
  if (APIkey == null) {
    returnFailure(res, "Provide the API key as an environment variable");
    return;
  }
```

Here are a few resources on this pattern:

* https://softwareengineering.stackexchange.com/questions/18454/should-i-return-from-a-function-early-or-use-an-if-statement
* https://dev.to/jpswade/return-early-12o5

Reply:
Thanks for the information :)